mongo/pagination/pagination_test.go

```diff
@@ -338,6 +343,59 @@ func TestPaginationPaginate(t *testing.T) {
 			},
 			ExpectedQuery: bson.M{"virtual_storage_name": "vs_name_2"},
 		},
+		{
+			Name: "it should not return deleted documents by default",
+			DummyDocument: func(t *testing.T) func() {
+				clean := newDummyDocumentsIncludingDeleted(t, "vs_name_1", 2, 1)
+				return func() {
+					clean()
+				}
+			},
+			ExpectedMeta: func() Meta {
+				return Meta{
+					CurrentPage: 1,
+					PrevPage:    nil,
+					NextPage:    nil,
+					TotalPages:  1,
+					TotalCount:  1,
+					perPageNum:  1,
+				}
+			},
+			ExpectedResult: []dummyDocument{
+				{AppID: "1", VirtualStorageName: "vs_name_1"},
+			},
+			ExpectedQuery: bson.M{"virtual_storage_name": "vs_name_1"},
+		},
+		{
+			Name: "it should not return deleted documents by default, with custom QueryFunc to document.WhereQueryUnscoped",
+			DummyDocument: func(t *testing.T) func() {
+				clean := newDummyDocumentsIncludingDeleted(t, "vs_name_1", 2, 1)
+				return func() {
+					clean()
+				}
+			},
+			PaginationOpts: &ServiceOpts{
+				PerPageDefault: 10,
+				MaxPerPage:     10,
+			},
+			AmountItems: 10,
+			ExpectedMeta: func() Meta {
+				return Meta{
+					CurrentPage: 1,
+					PrevPage:    nil,
+					NextPage:    nil,
+					TotalPages:  1,
+					TotalCount:  2,
+					perPageNum:  10,
+				}
+			},
+			ExpectedResult: []dummyDocument{
+				{AppID: "0", VirtualStorageName: "vs_name_1"},
+				{AppID: "1", VirtualStorageName: "vs_name_1"},
+			},
+			ExpectedQuery: bson.M{"virtual_storage_name": "vs_name_1"},
+			ExpectedQueryFunc: document.WhereUnscopedQuery,
+		},
```

Comment:
Suggestion: Add extra tests to verify the behaviour of the query when used in combination with PageNumber, AmountItems and SortOrder properties.

Reply:
Done